### ISSUE TITLE
Sentinel 5 second watch retry could miss updates

### ISSUE THREAD

**Issue by mdkent** [BlueDragonX/sentinel#9]
In tracking down these log entries:
```
sentinel 2015/01/16 13:34:52.429339 [ERROR]  watch on '/apps/foo/staging/_trigger' failed: unexpected end of JSON input
sentinel 2015/01/16 13:34:52.429354 [INFO]   retrying in 5s
sentinel 2015/01/16 13:34:52.429410 [ERROR]  watch on '/apps/foo/staging/_restart' failed: unexpected end of JSON input
sentinel 2015/01/16 13:34:52.429420 [INFO]   retrying in 5s
sentinel 2015/01/16 13:39:57.432530 [ERROR]  watch on '/apps/foo/staging/_restart' failed: unexpected end of JSON input
sentinel 2015/01/16 13:39:57.432545 [INFO]   retrying in 5s
sentinel 2015/01/16 13:39:57.432605 [ERROR]  watch on '/apps/foo/staging/_trigger' failed: unexpected end of JSON input
sentinel 2015/01/16 13:39:57.432616 [INFO]   retrying in 5s
sentinel 2015/01/16 13:45:02.435467 [ERROR]  watch on '/apps/foo/staging/_trigger' failed: unexpected end of JSON input
sentinel 2015/01/16 13:45:02.435484 [INFO]   retrying in 5s
sentinel 2015/01/16 13:45:02.435554 [ERROR]  watch on '/apps/foo/staging/_restart' failed: unexpected end of JSON input
sentinel 2015/01/16 13:45:02.435566 [INFO]   retrying in 5s
```
I've come to learn about the 5 minute etcd server-side watch timeout, which in itself is fine, but with the 5 second delay on the retry we could possibly miss some updates during the sleep.

I've confirmed this by extending the WatchRetry  (https://github.com/BlueDragonX/sentinel/blob/master/listener.go#L11) to 60 seconds and updating watched keys during the sleep, sentinel never sees the updates.

```
sentinel 2015/01/16 18:07:13.634183 [ERROR]  watch on '/apps/foo/staging/_restart' failed: unexpected end of JSON input
sentinel 2015/01/16 18:07:13.634199 [INFO]   retrying in 60s
sentinel 2015/01/16 18:07:13.634295 [ERROR]  watch on '/apps/foo/staging/_trigger' failed: unexpected end of JSON input
sentinel 2015/01/16 18:07:13.634306 [INFO]   retrying in 60s
[watched keys touched]
```

It'd be nicer to handle this with either an immediate reconnect then a gradual backoff to say 5 seconds, or possibly trying to interpret the empty reply from etcd when it closes the connection and doing an immediate retry. node-etcd for example parses the reply to distinguish between legit errors and closed connections: https://github.com/stianeikeland/node-etcd/blob/master/src/watcher.coffee#L76-L88. The added benefit to this approach would be logging the cause of the reconnect versus 'failed: unexpected end of JSON input' every 5 minutes into eternity.

**Comment by BlueDragonX** [comment]
That's for reporting this. It may be that the go-etcd library is causing some of these issues.

**Comment by BlueDragonX** [comment]
Fixed in v2.0.0.
